src/lib/setup.js

```diff
@@ -0,0 +1,65 @@
+import spawn from 'cross-spawn-promise';
+import { hasCommand, warn } from '../util';
+
+export function install(cwd, isYarn) {
+	let cmd = isYarn ? 'yarn' : 'npm';
+	return spawn(cmd, ['install'], { cwd, stdio: 'ignore' });
+}
+
+export async function addScripts(obj, cwd, isYarn) {
+	return {
+		build: 'zimlet build',
+		watch: 'zimlet watch',
+		package: 'zimlet package'
+	};
+}
+
+// Initializes the folder using `git init` and a proper `.gitignore` file
+// if `git` is present in the $PATH.
+export async function initGit(target) {
+	let git = hasCommand('git');
+
+	if (git) {
+		const cwd = target;
+
+		await spawn('git', ['init'], { cwd });
+		await spawn('git', ['add', '-A'], { cwd });
+
+		let gitUser, gitEmail;
+		const defaultGitUser = 'Zimlet CLI';
+		const defaultGitEmail = 'zimlet-cli@users.noreply.github.com';
+
+		try {
+			gitEmail = (await spawn('git', ['config', 'user.email'])).toString();
+		}
+		catch (e) {
+			gitEmail = defaultGitEmail;
+		}
+
+		try {
+			gitUser = (await spawn('git', ['config', 'user.name'])).toString();
+		}
+		catch (e) {
+			gitUser = defaultGitUser;
```

Comment:
Is it too verbose to warn the user when we fallback to default git email/username values?

Reply:
no, I think that is a good warning to have.  I'll put it in